core/embed/rust/rust_smp.h

```diff
@@ -2,10 +2,24 @@
 
 #include <trezor_types.h>
 
+/// Represents a parsed version string in the format:
```

Comment:
thanks for adding comments here, please add it for the rest of functions here too, make it doxygen style and add license header while at it